modules/updater/updater.lua

```diff
@@ -165,35 +163,40 @@ local function updateFiles(data, keepCurrentFiles)
       end
     end
   end
-  
+
   updaterWindow.status:setText(tr("Updating %i files", #toUpdate))
   updaterWindow.mainProgress:setPercent(0)
   updaterWindow.downloadProgress:setPercent(0)
   updaterWindow.downloadProgress:show()
   updaterWindow.downloadStatus:show()
   updaterWindow.changeUrlButton:hide()
+
   downloadFiles(data["url"], toUpdate, 1, 0, function()
     updaterWindow.status:setText(tr("Updating client (may take few seconds)"))
     updaterWindow.mainProgress:setPercent(100)
     updaterWindow.downloadProgress:hide()
-    updaterWindow.downloadStatus:hide() 
+    updaterWindow.downloadStatus:hide()
     scheduledEvent = scheduleEvent(function()
       local restart = binary or (not loadModulesFunction and reloadModules) or forceRestart
       if newFiles then
-        g_resources.updateFiles(finalFiles, not restart)
+        g_resources.updateFiles(toUpdateFiles, not restart)
       end
+
       if binary then
         g_resources.updateExecutable(binary)
       end
+
       if restart then
-        g_app.restart()
+        g_logger.info("restart")
+        --g_app.exit()
+        -- TODO Restart
```

Comment:
I have something like this
```
void Application::restart()
{
#if not(defined(ANDROID))
    STARTUPINFO si;
    PROCESS_INFORMATION pi;
    ZeroMemory(&si, sizeof(si));
    si.cb = sizeof(si);
    ZeroMemory(&pi, sizeof(pi));

    // Create the child process.
    if (CreateProcess((LPCSTR)g_resources.getBinaryName().c_str(), // path to the executable
        nullptr, // command line arguments
        nullptr, // process handle not inheritable
        nullptr, // thread handle not inheritable
        FALSE, // handle inheritance flag
        0, // creation flags
        nullptr, // use parent's environment block
        nullptr, // use parent's starting directory
        &si, // pointer to STARTUPINFO structure
        &pi) // pointer to PROCESS_INFORMATION structure
    ) {
        // Close process and thread handles.
        CloseHandle(pi.hProcess);
        CloseHandle(pi.hThread);
        
    } else {
        g_logger.fatal("Updater restart error. Please restart application");
    }
    quick_exit();
#else
    exit();
#endif
}
```

Reply:
I will try and let you know, I still need to figure out how to properly update the .exe file

Reply:
@conde2 Does this work also for linux/mac executables?

Reply:
The updater or this code from poxis?
The updater should work for mac/linux not sure the if the executable will update, I use windows so not able to test in other systems.

You can try removing the FREE_VERSION and test
```cpp
void ResourceManager::updateExecutable(std::string fileName)
{
#if defined(ANDROID) || defined(FREE_VERSION)
    g_logger.fatal("Executable cannot be updated on android or in free version");
```